shared/analytics_tracking/events.py

```diff
@@ -7,3 +18,24 @@ class Events(Enum):
     ACCOUNT_UPLOADED_COVERAGE_REPORT = "Account Uploaded Coverage Report"
     USER_SIGNED_IN = "User Signed In"
     USER_SIGNED_UP = "User Signed Up"
+
+
+class Event:
+    def __init__(self, name: str, dt: datetime = None, **data: Any) -> None:
```

Comment:
Shouldn't `name` be of type `Events`? (which might be called `EventName` possibly)

So that I can pass for example `name=Events.USER_SIGNED_IN` directly?

Reply:
good point. The reason for this is that I don't want to let the manager worry about valid event names whatsoever, it only passes the string that the client code sends. Then the name gets validated in the Event class

Reply:
I would argue that using an Enum still counts as "the manager doesn't have to worry about event names". It's the user - calling `track_event` - that needs to worry about having a valid event name and what that event name should be :E 

That being said I think your reasoning is valid and if you want to keep as is it's fine too.